lib/galaxy/datatypes/binary.py

```diff
@@ -1854,12 +1854,66 @@ def set_meta(self, dataset, overwrite=True, **kwd):
             log.warning("%s, set_meta Exception: %s", self, e)
 
     def sniff(self, filename):
-        if super(IdpDB, self).sniff(filename):
+        if super(GAFASQLite, self).sniff(filename):
             table_names = frozenset({'gene', 'gene_family', 'gene_family_member', 'meta', 'transcript'})
             return self.sniff_table_names(filename, table_names)
         return False
 
 
+class NcbiTaxonomySQlite(SQlite):
+    """Class describing the NCBI Taxonomy database stored in SQLite as done by rust-ncbitaxonomy"""
+    MetadataElement(name='ncbitaxonomy_schema_version', default='20200501095116', param=MetadataParameter, desc='ncbitaxonomy schema version',
+                    readonly=True, visible=True, no_value='20200501095116')
+    MetadataElement(name="taxon_count", default=[], param=MetadataParameter, desc="Count of taxa in the taxonomy",
+                    readonly=True, visible=True, no_value=[])
+
+    file_ext = 'ncbitaxonomy.sqlite'
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        super(NcbiTaxonomySQlite, self).set_meta(dataset, overwrite=overwrite, **kwd)
+        try:
+            conn = sqlite.connect(dataset.file_name)
+            c = conn.cursor()
+            version_query = 'SELECT version FROM __diesel_schema_migrations ORDER BY run_on DESC LIMIT 1'
+            results = c.execute(version_query).fetchall()
+            if len(results) == 0:
+                raise Exception('version not found in __diesel_schema_migrations table')
+            dataset.metadata.ncbitaxonomy_schema_version = results[0][0]
+            taxons_query = 'SELECT count(name) FROM taxonomy'
+            results = c.execute(taxons_query).fetchall()
+            if len(results) == 0:
+                raise Exception('could not count size of taxonomy table')
+            dataset.metadata.taxon_count = results[0][0]
+        except Exception as e:
+            log.warning("%s, set_meta Exception: %s", self, e)
+
+    def sniff(self, filename):
+        if super(NcbiTaxonomySQlite, self).sniff(filename):
+            table_names = frozenset({'__diesel_schema_migrations', 'taxonomy'})
```

Comment:
Is it exactly these two tables? I guess one wants the check as specific as possible. But I admit that `__diesel_schema_migrations` seems quite specific .. is it?

Reply:
Yes it is exactly these 2 tables. The funnily named `__diesel_schema_migrations` table is used by the Rust diesel interface to keep track of schema versions and migrations.

Reply:
Yes, the `__diesel_schema_migrations` table stores the version of the schema and info on migrations from the Rust diesel ORM. And then the `taxonomy` table stores the data.